### PR TITLE
issue 42 fix

### PR DESCRIPTION
**Pull request by modernpacifist**

### CODE REVIEW

File: sisulib.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import csv
 import json
 import os
-
 import airtable
+import hashlib
+
+from rhinoscriptsyntax import DocumentPath
```
Comment (tmshv) [inline sisulib.py:7]: No `rhinoscriptsyntax` in this file. We were talking about it.

**Review by tmshv — CHANGES_REQUESTED** [review]

File: sisulib.py [hunk 2]
```diff
@@ -160,3 +162,14 @@ def get_provider(config):
         return airtable.AirtableProvider(options)
 
     return None
+
+
+def get_hash(sisu_config):
+
+    rhinofile_path = get_rhinofile_path()
```
Comment (tmshv) [inline sisulib.py:169]: You created transitive dependency of `rhinoscriptsyntax`. Nothing changed.

**Review by tmshv — CHANGES_REQUESTED** [review]

File: sisulib.py (hunk 2 above)
Comment (modernpacifist) [reply sisulib.py:169]: Is there a way to get Rhino file without `rhinoscriptsyntax`?

Comment (tmshv) [reply sisulib.py:169]: Pass it to hash function as argument. We were discussed about it.

Comment (modernpacifist) [reply sisulib.py:169]: - get_hash() is a method in Sisuconfig classes
- takes kwargs as argument
- sorts them in order and returns md5 hash

**Comment by modernpacifist** [comment]
The version of the script will be passed as a parameter or must be parsed from __plugin__.py?

File: sisulib.py [hunk 3]
```diff
@@ -23,6 +25,17 @@ def full_layer_name(layer, parent):
                 layer_names.append(full_layer_name(view_name, name))
         return layer_names
 
+    def get_hash(self, **kwargs):
+        sisufile_path = os.path.join(kwargs.get("filename", "sisufile.json"))
+
+        with open(sisufile_path, "r") as f:
+            sisufile_content = "".join(map(str, f.readlines()))
```
Comment (tmshv) [inline sisulib.py:32]: why not just `f.read()`?

File: sisulib.py [hunk 4]
```diff
@@ -23,6 +25,17 @@ def full_layer_name(layer, parent):
                 layer_names.append(full_layer_name(view_name, name))
         return layer_names
 
+    def get_hash(self, **kwargs):
```
Comment (tmshv) [inline sisulib.py:28]: add tests

**Review by tmshv — CHANGES_REQUESTED** [review]

File: sisulib.py [hunk 5]
```diff
@@ -23,6 +25,20 @@ def full_layer_name(layer, parent):
...
+        sisufile_path = os.path.join(kwargs.get("filename", "sisufile.json"))
+
+        with open(sisufile_path, "r") as f:
+            sisufile_content = f.read()
+
+        if not kwargs.get("version"):
+            kwargs.update({"version": version})
+
+        values = [str(kwargs[key]) for key in sorted(kwargs.keys())]
+        values_string = "".join(values)
```
Comment (tmshv) [inline sisulib.py:38]: make it more pythonic:

```py
values = "".join([
    str(v)
    for k, v in sorted(kwargs.items(), lambda k, v: k)
])
```

or:

```py
values = "".join([
    str(kwargs[key])
    for key in sorted(kwargs)
])
```

**Review by tmshv — CHANGES_REQUESTED** [review]

**Review by tmshv — APPROVED** [review]

**Pull request merged**